src/ansys/geometry/core/_grpc/_services/v1/patterns.py

```diff
@@ -43,30 +49,205 @@ class GRPCPatternsServiceV1(GRPCPatternsService):  # pragma: no cover
 
     @protect_grpc
     def __init__(self, channel: grpc.Channel):  # noqa: D102
-        from ansys.api.geometry.v1.patterns_pb2_grpc import PatternsStub
+        from ansys.api.discovery.v1.design.relationships.pattern_pb2_grpc import PatternStub
 
-        self.stub = PatternsStub(channel)
+        self.stub = PatternStub(channel)
 
     @protect_grpc
     def create_linear_pattern(self, **kwargs) -> dict:  # noqa: D102
-        raise NotImplementedError
+        from ansys.api.discovery.v1.design.relationships.pattern_pb2 import (
+            LinearPatternCreationRequest,
+            LinearPatternCreationRequestData,
+        )
+
+        # Create the request - assumes all inputs are valid and of the proper type
+        request = LinearPatternCreationRequest(
+            request_data=[
+                LinearPatternCreationRequestData(
+                    selection_ids=[build_grpc_id(id) for id in kwargs["selection_ids"]],
+                    linear_direction_id=build_grpc_id(kwargs["linear_direction_id"]),
+                    count_x=kwargs["count_x"],
+                    pitch_x=from_length_to_grpc_quantity(kwargs["pitch_x"]),
+                    two_dimensional=kwargs["two_dimensional"],
+                    count_y=kwargs["count_y"],
+                    pitch_y=(
+                        from_length_to_grpc_quantity(kwargs["pitch_y"])
+                        if kwargs["pitch_y"]
+                        else from_length_to_grpc_quantity(Distance(0))
+                    ),
```

Comment:
Shouldn't this come from the call itself? I would rather avoid any logic inside the _grpc layer